tonic/src/transport/channel/service/add_origin.rs

```diff
@@ -54,8 +54,8 @@ where
             // Split the request URI into parts.
             let mut uri: http::uri::Parts = head.uri.into();
             // Update the URI parts, setting hte scheme and authority
-            uri.scheme = self.scheme.clone();
-            uri.authority = self.authority.clone();
```

Comment:
I think these fixes are good. 

As `assigning_clones` rule will be moved to `pedantic` and is scheduled to be allowed by default, when we adopt this rule, setting it as `warn` might be desirable.

https://github.com/rust-lang/rust-clippy/pull/12779